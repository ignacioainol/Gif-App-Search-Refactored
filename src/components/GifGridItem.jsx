import { removeWord } from "../helpers/globalFunctions"

export const GifGridItem = ({ title, url }) => {

  const formatTitle = removeWord(title, 'GIF');

  return (
    <div className="col-xs-12 col-sm-3 card">
      < img src={url} alt={title} />
      <p>{formatTitle}</p>
    </div >
  )
}
