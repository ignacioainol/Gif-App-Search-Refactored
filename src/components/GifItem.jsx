import { removeWord } from "../helpers/globalFunctions"
import PropTypes from 'prop-types';


export const GifItem = ({ title, url }) => {

  const formatTitle = removeWord(title, 'GIF');

  return (
    <div className="col-xs-12 col-sm-3 card">
      < img src={url} alt={title} />
      <p>{formatTitle}</p>
    </div >
  )
}

GifItem.propTypes = {
  title: PropTypes.string.isRequired,
  url: PropTypes.string.isRequired
};