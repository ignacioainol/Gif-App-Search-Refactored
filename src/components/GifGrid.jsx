import { GifItem } from './GifItem';
import { useFetchGifs } from '../hooks/useFetchGifs';
import { Loading } from './Loading';

export const GifGrid = ({ category }) => {

  const { images, isLoading } = useFetchGifs(category);

  return (
    <>
      <h3>{category}</h3>

      {isLoading &&
        <Loading />
      }

      <div className='row'>
        {
          images.map((image) => (
            <GifItem
              key={image.id}
              {...image}
            />
          ))
        }
      </div>
    </>
  )
}
