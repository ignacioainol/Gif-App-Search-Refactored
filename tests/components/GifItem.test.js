import { render, screen } from "@testing-library/react"
import { GifItem } from "../../src/components/GifItem"

describe('Test in GifItem', () => {

  const title = 'Saitama';
  const url = 'https://one-punch..com/saitama.jpeg';

  test('Debe hacer match con el snapshot', () => {

    const { container } = render(<GifItem title={title} url={url} />);
    expect(container).toMatchSnapshot();
  });

  test('Debe de mostrar la imagen con el URLy el ALT indicado', () => {
    render(<GifItem title={title} url={url} />);
    // screen.debug();
    // expect(screen.getByRole('img').src).toBe(url);
    const { src, alt } = screen.getByRole('img');
    expect(src).toBe(url)
    expect(alt).toBe(title)

  });

  test('Debe de mostrar el title en el component', () => {
    render(<GifItem title={title} url={url} />);
    expect(screen.getByText(title)).toBeTruthy();
  })
})