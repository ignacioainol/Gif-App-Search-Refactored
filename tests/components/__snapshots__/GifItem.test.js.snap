// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test in GifItem Debe hacer match con el snapshot 1`] = `
<div>
  <div
    class="col-xs-12 col-sm-3 card"
  >
    <img
      alt="Saitama"
      src="https://one-punch..com/saitama.jpeg"
    />
    <p>
      Saitama
    </p>
  </div>
</div>
`;
